src/components/organisms/dashboard/templates/tabs/multiplePopoverSelect.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useMemo, useState, useEffect, useRef } from "react";
+
+import { defaultSelectedMultipleSelect } from "@src/constants";
+import { MultiplePopoverSelectProps } from "@src/interfaces/components";
+
+import { Typography } from "@components/atoms";
+import { PopoverListContent, PopoverListTrigger, PopoverListWrapper } from "@components/molecules/popover";
+import { MultipleLabelPopoverItem } from "@components/organisms/dashboard/templates/tabs";
+
+import { ChevronDownIcon, Close } from "@assets/image/icons";
+
+export const MultiplePopoverSelect = ({
+	label,
+	items,
+	emptyListMessage,
+	defaultSelectedItems = [],
+	onItemsSelected,
+}: MultiplePopoverSelectProps) => {
+	const [selectedItem, setSelectedItem] = useState<string[]>(
+		defaultSelectedItems.length > 0 ? defaultSelectedItems : [defaultSelectedMultipleSelect]
+	);
+	const [showCloseIcon, setShowCloseIcon] = useState(
+		defaultSelectedItems.length > 0 && !defaultSelectedItems.includes(defaultSelectedMultipleSelect)
+	);
+	const containerRef = useRef<HTMLDivElement>(null);
+	const [contentWidth, setContentWidth] = useState<number | undefined>(undefined);
+
+	useEffect(() => {
+		if (defaultSelectedItems.length === 0) {
+			onItemsSelected?.([defaultSelectedMultipleSelect]);
+		}
+	}, [defaultSelectedItems, onItemsSelected]);
+
+	useEffect(() => {
+		const resizeObserver = new ResizeObserver((entries) => {
+			for (const entry of entries) {
+				setContentWidth(entry.contentRect.width);
+			}
+		});
+
+		const currentContainerRef = containerRef.current;
+
+		if (currentContainerRef) {
+			resizeObserver.observe(currentContainerRef);
+		}
+
+		return () => {
+			if (currentContainerRef) {
+				resizeObserver.unobserve(currentContainerRef);
+			}
+		};
+	}, []);
+
+	const handleItemSelect = ({ id }: { id: string }) => {
+		setSelectedItem((prevSelected) => {
+			if (id === defaultSelectedMultipleSelect) {
+				onItemsSelected?.([defaultSelectedMultipleSelect]);
+				setShowCloseIcon(false);
+				return [defaultSelectedMultipleSelect];
+			}
+
+			const isSelected = prevSelected.includes(id);
+			const filteredItems = prevSelected.filter((item) => item !== defaultSelectedMultipleSelect && item !== id);
+
+			let newSelected: string[];
+			if (isSelected && filteredItems.length === 0) {
+				newSelected = [defaultSelectedMultipleSelect];
+			} else if (isSelected) {
+				newSelected = filteredItems;
+			} else {
+				newSelected = [...filteredItems, id];
+			}
+
+			onItemsSelected?.(newSelected);
+			setShowCloseIcon(newSelected.length > 0 && !newSelected.includes(defaultSelectedMultipleSelect));
+			return newSelected;
+		});
+	};
```

Comment:
The MultipleLabelPopoverItem component currently creates new functions on each render. You can optimize it by memoizing the event handlers:

```suggestion
	const handleItemSelect = useCallback(({ id }: { id: string }) => {
		setSelectedItem((prevSelected) => {
			if (id === defaultSelectedMultipleSelect) {
				onItemsSelected?.([defaultSelectedMultipleSelect]);
				setShowCloseIcon(false);
				return [defaultSelectedMultipleSelect];
			}

			const isSelected = prevSelected.includes(id);
			const filteredItems = prevSelected.filter((item) => item !== defaultSelectedMultipleSelect && item !== id);

			let newSelected: string[];
			if (isSelected && filteredItems.length === 0) {
				newSelected = [defaultSelectedMultipleSelect];
			} else if (isSelected) {
				newSelected = filteredItems;
			} else {
				newSelected = [...filteredItems, id];
			}

			onItemsSelected?.(newSelected);
			setShowCloseIcon(newSelected.length > 0 && !newSelected.includes(defaultSelectedMultipleSelect));
			return newSelected;
		});
	}, [onItemsSelected, defaultSelectedMultipleSelect]);
```

Reply:
changed